thesis-fe/src/components/container/AddDestination.js

```diff
@@ -0,0 +1,106 @@
+import React, { Component } from 'react';
+import styled from 'react-emotion'
+import { fontFamily } from '../../helpers/constants';
+import { WizardMode } from './WizardMode';
+import { List } from './List';
+
+
+export class AddDestination extends Component {
+  state = {
+    input: '',
+    dictator: false,
+  }
+
+  componentDidMount() {
+    if (this.props.destination.chosenOne) this.setState({ dictator: true});
+    this.props.destination.chosenOne &&
+    this.setState({ input: this.props.destination.chosenOne });
+  }
+
+  handleInput = async (event) => {
+    let chosenOne;
+    await this.setState({ input: event.target.value });
+    if (this.state.dictator) {
+      this.state.input.length ? chosenOne = this.state.input : chosenOne = null;
+      this.props.setDestination({ suggestions: [], chosenOne });
+    }
+  }
+
+  handleAddClick = async () => {
+    const parentSuggestions = this.props.destination.suggestions;
+    const destinationInput = this.state.input;
+    let suggestions;
+    if (parentSuggestions){
+      if (parentSuggestions.includes(destinationInput)) return;
+      suggestions = parentSuggestions.slice();
+    } else suggestions = [];
+    suggestions.push(destinationInput);
+    await this.setState({ input: '' });
+    this.props.setDestination({ suggestions, chosenOne: null});
+  }
+
+  setMode = async (flag) => {
+    await this.setState({ dictator: flag });
+  }
```

Comment:
This doesn't even make sense. Apart from `this.setState` not returning a promise, if it would you're wrapping a function that returns a promise with another promise.